forge/test/operators/pytorch/tm/test_reshape.py

```diff
@@ -0,0 +1,178 @@
+    
+    while True:
+        num_dims = random.randint(1, max_dims)  # Limit the maximum number of dimensions
+        
+        # Attempt to divide total_elements into a random number of factors
+        target_shape = []
+        remaining = total_elements
+        for i in range(num_dims - 1):
+            if remaining == 1:
+                target_shape.extend([1] * (num_dims - len(target_shape)))
+                break
+            dim_size = random.randint(1, remaining)
+            while remaining % dim_size != 0:  # Ensure that dim_size divides remaining
+                dim_size = random.randint(1, remaining)
+            target_shape.append(dim_size)
+            remaining //= dim_size
+        
+        target_shape.append(remaining)  # Add the final dimension
+        random.shuffle(target_shape)  # Shuffle the dimensions for additional randomness
+        
+        if tuple(target_shape) != source_shape and len(target_shape) <= max_dims:  # Ensure target shape is valid
+            break
+        
+        num_attempts += 1
+        if num_attempts > 100:  # Prevent infinite loop if no valid shape can be found
+            raise ValueError(f"Failed to generate a target shape different from {source_shape} after 100 attempts.")
+    
+    return tuple(target_shape)
+
+
+class TestVerification:
+
+    MODEL_TYPES = {
+        InputSource.FROM_ANOTHER_OP: ModelFromAnotherOp,
+        InputSource.FROM_HOST: ModelDirect,
+        InputSource.FROM_DRAM_QUEUE: ModelDirect,
+        InputSource.CONST_EVAL_PASS: ModelConstEvalPass,
+    }
+
+    @classmethod
+    def verify(
+        cls,
+        test_device: TestDevice,
+        test_vector: TestVector,
+        input_params: List[Dict] = [],
+        warm_reset: bool = False,
+    ):
+
+        input_source_flag: InputSourceFlags = None
+        if test_vector.input_source in (InputSource.FROM_DRAM_QUEUE,):
+            input_source_flag = InputSourceFlags.FROM_DRAM
+
+        operator = getattr(torch, test_vector.operator)
+        kwargs = test_vector.kwargs if test_vector.kwargs else {}
+
+        model_type = cls.MODEL_TYPES[test_vector.input_source]
+        pytorch_model = (
+            model_type(operator, test_vector.input_shape, kwargs)
+            if test_vector.input_source in (InputSource.CONST_EVAL_PASS,)
+            else model_type(operator, kwargs)
+        )
+
+        input_shapes = tuple([test_vector.input_shape])
+
+        logger.trace(f"***input_shapes: {input_shapes}")
+
+        VerifyUtils.verify(
+            model=pytorch_model,
+            test_device=test_device,
+            input_shapes=input_shapes,
+            input_params=input_params,
+            input_source_flag=input_source_flag,
+            dev_data_format=test_vector.dev_data_format,
+            math_fidelity=test_vector.math_fidelity,
+            pcc=test_vector.pcc,
+            warm_reset=warm_reset,
+            value_range=ValueRanges.SMALL,
+        )
```

Comment:
verify that output shape is same as target shape

Reply:
Now we use new verify module that checks this.